background.ts

```diff
@@ -749,17 +766,63 @@ function newK8sManager() {
 
 /**
  * Implement the methods the HttpCommandServer needs to service its requests.
- * These methods should be thin wrappers around existing functionality in the rest of the backend.
- * Getters normally return strings, either JSON strings or scalars.
- * Setters normally return either `true | false`, or possibly `true | error`.
- * The `requestShutdown` is a special case that never returns.
+ * These methods do two things:
+ * 1. Verify the semantics of the parameters (the server just checks syntax).
+ * 2. Provide a thin wrapper over existing functionality in this module.
+ * Getters, on success, return status 200 and a string that may be JSON or simple.
+ * Setters, on success, return status 202, possibly with a human-readable status note.
+ * The `requestShutdown` method is a special case that never returns.
  */
 class BackgroundCommandWorker implements CommandWorkerInterface {
+  protected k8sVersions: string[] = [];
+  protected settingsValidator = new SettingsValidator(cfg);
+
   getSettings() {
     return JSON.stringify(cfg, undefined, 2);
   }
 
-  requestShutdown() {
+  /**
+   * Check semantics of SET commands:
+   * - verify that setting names are recognized, and validate provided values
+   * - returns an array of two strings:
+   *   1. a description of the status of the request, if it was valid
+   *   2. a list of any errors in the request body.
+   * @param newSettings: a subset of the Settings object, containing the desired values
+   * @returns [{string} description of final state if no error, {string} error message]
+   */
+  async updateSettings(newSettings: Record<string, any>): Promise<[string, string]> {
+    if (this.k8sVersions.length === 0) {
+      this.k8sVersions = (await k8smanager.availableVersions).map(entry => entry.version.version);
+      this.settingsValidator.k8sVersions = this.k8sVersions;
+    }
+    this.settingsValidator.cfg = cfg;
```

Comment:
This code is now obsolete, but I did it this way because `cfg` is a volatile module-global variable. But you're right, no need to also assign it in the constructor. Both forms are gone now.